ch4/deepComparison.js

```diff
@@ -0,0 +1,42 @@
+const deepEqual = (value1, value2) => {
+  if (value1 === value2) return true;
+  if (value1 === null || value2 === null) 
+  if (typeof value1 !== typeof value2) return false;
+
+  if(typeof value1 === 'object') {
+    if (Object.keys(value1).length !== Object.keys(value2).length) {
+        return false;
+    }
+
+    Object.keys(value1).forEach((key) => {
+      if (!(Object.keys(value2).includes(key)) ||
+        !deepEqual(value1[key],value2[key])) return false;
```

Comment:
```suggestion
        !deepEqual(value1[key], value2[key])) return false;
```
add a space between args